flexmeasures/data/models/planning/storage.py

```diff
@@ -423,6 +383,111 @@ def ensure_soc_min_max(self):
                     )
 
 
+class StorageFallbackScheduler(MetaStorageScheduler):
+
+    __version__ = "1"
+    __author__ = "Seita"
+
+    def compute(self, skip_validation: bool = False) -> pd.Series | None:
+        """Schedule a battery or Charge Point by just starting to charge, discharge, or do neither,
+           depending on the first target state of charge and the capabilities of the Charge Point.
+           Note that this ignores any cause of the infeasibility.
+
+        :param skip_validation: If True, skip validation of constraints specified in the data.
+        :returns:               The computed schedule.
+        """
+
+        (
+            sensor,
+            start,
+            end,
+            resolution,
+            soc_at_start,
+            device_constraints,
+            ems_constraints,
+            commitment_quantities,
+            commitment_downwards_deviation_price,
+            commitment_upwards_deviation_price,
+        ) = self._prepare(skip_validation=skip_validation)
+
+        # Fallback policy if the problem was unsolvable
+        battery_schedule = fallback_charging_policy(
+            sensor, device_constraints[0], start, end, resolution
+        )
+
+        # Round schedule
+        if self.round_to_decimals:
+            battery_schedule = battery_schedule.round(self.round_to_decimals)
+
+        return battery_schedule
+
+
+class StorageScheduler(MetaStorageScheduler):
+    """Schedule a battery or Charge Point based directly on the latest beliefs regarding market prices within the specified time window.
+    For the resulting consumption schedule, consumption is defined as positive values
+    """
+
+    __version__ = "3"
+    __author__ = "Seita"
+
+    fallback_scheduler_class: Scheduler = StorageFallbackScheduler
+
+    @property
+    def fallback_scheduler(self):
+        _fallback_scheduler = self.fallback_scheduler_class(
+            self.sensor,
+            self.start,
+            self.end,
+            self.resolution,
+            self.belief_time,
+            self.round_to_decimals,
+            self.flex_model,
+            self.flex_context,
```

Comment:
Args sometimes make me nervous (especially when instantiating a class). Can we use kwargs here instead?

Reply:
Yes haha it makes me anxious as well. Actually, I end up removing this property...